### PR TITLE
Add pagination demo

### PR DESCRIPTION
**Pull request by CaroFG**
- [X] Need to update instant-meilisearch dependency when new version (not beta) is released

### CODE REVIEW

File: src/numbered-pagination/src/style.css [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+  button.btn.btn-dodger-blue.btn-lg {
+    align-self: flex-start;
+  }
+  .search-header-content {
+    padding-left: 5rem;
+    padding-right: 5rem;
+  }
+  #logo {
+    background-image: url(/meili_logo.svg);
+  }
```
Comment (mdubus) [inline src/numbered-pagination/src/style.css:169]: The logo shouldn't be in the media query :) It's disappearing in mobile 🙈

**Review by mdubus — COMMENTED** [review]
Amazing work 🤩 👏

**Review by oluademola — COMMENTED** [review]
Thank you for this fantastic demo. Apart from the changes requested by Morgane, everything looks good to me.

File: src/numbered-pagination/README.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+This demo showcases Meilisearch's v0.30 new pagination mode: numbered pagination.
+
+v0.30 introduces a numbered pagination mode for users who need pagination interfaces.
+
+You can find more information about this pagination mode in [our documentation](https://docs.meilisearch.com/learn/advanced/pagination.html#numbered-page-selectors)
+
+If you want to replicate this demo using your own Meilisearch instance, you have to follow these steps:
+  
+### 1. Downloand and launch Meilisearch
```
[the remark this answers is not in the document]
Comment (CaroFG) [reply src/numbered-pagination/README.md:34]: Oh god xD

File: src/numbered-pagination/src/style.css (hunk 1 above)
Comment (CaroFG) [reply src/numbered-pagination/src/style.css:169]: That's weird it disappears! I have a logo for the regular page and one with just the logo mark for mobiles... I did the exact same thing on the matching strategy demo, and it works perfectly... O_o

**Review by mdubus — APPROVED** [review]
LGTM ✨🦕

**Pull request merged**